cwbi-auth-http-client/src/main/java/hec/army/usace/hec/cwbi/auth/http/client/DirectGrantX509TokenRequestBuilder.java

```diff
@@ -0,0 +1,56 @@
+package hec.army.usace.hec.cwbi.auth.http.client;
+
+import hec.army.usace.hec.cwbi.auth.http.client.token.fluentbuilders.DirectGrantX509TokenRequestFluentBuilder;
+import hec.army.usace.hec.cwbi.auth.http.client.token.fluentbuilders.TokenRequestFluentBuilder;
+import hec.army.usace.hec.cwbi.auth.http.client.trustmanagers.CwbiAuthTrustManager;
+import java.io.IOException;
+import java.util.Objects;
+import javax.net.ssl.SSLSocketFactory;
+import mil.army.usace.hec.cwms.http.client.ApiConnectionInfoBuilder;
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilderImpl;
+import mil.army.usace.hec.cwms.http.client.HttpRequestResponse;
+import mil.army.usace.hec.cwms.http.client.SslSocketData;
+import mil.army.usace.hec.cwms.http.client.auth.OAuth2Token;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestExecutor;
+
+public final class DirectGrantX509TokenRequestBuilder implements DirectGrantX509TokenRequestFluentBuilder {
+
+    private SslSocketData sslSocketData;
+
+    @Override
+    public TokenRequestFluentBuilder withSSlSocketFactory(SSLSocketFactory sslSocketFactory) {
+        this.sslSocketData = new SslSocketData(Objects.requireNonNull(sslSocketFactory, "Missing required SSLSocketFactory"),
+            CwbiAuthTrustManager.getTrustManager());
+        return new TokenRequestBuilderImpl();
+    }
+
+    private class TokenRequestBuilderImpl extends TokenRequestBuilder {
+
+        @Override
+        OAuth2Token retrieveToken() throws IOException {
+            OAuth2Token retVal = null;
+            String formBody = new UrlEncodedFormData()
```

Comment:
when we go to login.gov, the user/pass would be filled?

Reply:
yes, which I think would just be a DirectGrant authentication.